apps/web/pages/api/book/set-location.ts

```diff
@@ -0,0 +1,164 @@
+import dayjs from "dayjs";
+import { NextApiRequest, NextApiResponse } from "next";
+
+import EventManager from "@calcom/core/EventManager";
+import logger from "@calcom/lib/logger";
+import { getTranslation } from "@calcom/lib/server/i18n";
+import type { CalendarEvent } from "@calcom/types/Calendar";
+import type { AdditionInformation } from "@calcom/types/Calendar";
+
+import { getSession } from "@lib/auth";
+import { sendLocationChangeEmails } from "@lib/emails/email-manager";
+import prisma from "@lib/prisma";
+
+export default async function handler(req: NextApiRequest, res: NextApiResponse) {
+  if (req.method !== "POST") {
+    return res.status(405).json({ message: "Method not allowed" });
+  }
+
+  const { bookingId, newLocation: location } = req.body;
+
+  const session = await getSession({ req: req });
+
+  if (!session?.user?.id) {
+    res.status(401).json({ message: "Not authenticated" });
+    return;
+  }
+
+  try {
+    const currentUser = await prisma.user.findFirst({
+      where: {
+        id: session.user.id,
+      },
+      select: {
+        id: true,
+        credentials: true,
+        timeZone: true,
+        email: true,
+        name: true,
+        username: true,
+        destinationCalendar: true,
+        locale: true,
+      },
+    });
+
+    if (!currentUser) {
+      return res.status(401).json({ message: "Not authenticated" });
+    }
+
+    await prisma.booking.update({
+      where: {
+        id: bookingId,
+      },
+      data: {
+        location,
+      },
+    });
+
```

Comment:
Any logged in user can change the location of any booking knowing it's ID. This is a potential vulnerability.